src/storage/data_table.cpp

```diff
@@ -275,7 +280,7 @@ static void VerifyGeneratedExpressionSuccess(ClientContext &context, TableCatalo
 	try {
 		executor.ExecuteExpression(chunk, result);
 	} catch (std::exception &ex) {
-
+		// FIXME: should this be this lenient? assertions are also caught this way
```

Comment:
Perhaps we can have a separate path that catches `InternalException` and rethrows in that case?